Style

calebbraun opened the pull request:
This pull request adds on to PR #42, making several more changes to the general aesthetics of the package.

### Summary

- Attempts to create default maps with a cleaner style for easier focus on the data
- Adds a customizable background (ocean) polygon for increased contrast
- Adds option for padding between map edge and panel edge; this is done by default for world maps
- Uses darker default polygon color for increased contrast
- Removes graticules from default maps
- Adds graticule placement as a parameter, now allowing them to be placed over map polygons
- Makes zooming more intuitive, meaning negative zoom values now zoom out instead of in
- Handles more gracefully cases where reprojections create invalid polygons

Additionally, this PR groups map functions from one file into three separate files.


### **Map examples**

Old map with regional data:
![robworldold](https://user-images.githubusercontent.com/12344924/39065469-e54e0918-449f-11e8-84af-3ceacfc10522.png)
New map with regional data:
![robworldnew](https://user-images.githubusercontent.com/12344924/39065477-e92949b2-449f-11e8-8c39-0a3ecc6a2bcb.png)
Old map plotted with no data:
![nodataold](https://user-images.githubusercontent.com/12344924/39065399-b58141f0-449f-11e8-8e7b-a571f74a9657.png)
New map plotted with no data:
![nodatanew](https://user-images.githubusercontent.com/12344924/39065424-c7b0d9d0-449f-11e8-9f0a-37c73a2ed0c2.png)

calebbraun committed:
Remove depreciated functions

calebbraun committed:
Remove references to depreciated functions

calebbraun committed:
Remove commented code

calebbraun committed:
Remove excess global definitions

calebbraun committed:
Fix filtering error

- Remove old test code for spatial filtering
- Make lookup table for US consistent with others

resolves #36

calebbraun committed:
Make province translation/drop functions simpler

Addresses #31

calebbraun committed:
Begin resolving issues with using map.chn

calebbraun committed:
Finalize USA map

Update vignettes for China and USA

Resolves #35

calebbraun committed:
Accept basin fix from master

calebbraun committed:
Begin creating add_region_id() tests

calebbraun committed:
Resolve #41

calebbraun committed:
Fix test errors

calebbraun committed:
Attempt to allow more projections

calebbraun committed:
Filter after reprojection

calebbraun committed:
Try handling ortho in different ways

calebbraun committed:
Make extent the map bounds

calebbraun committed:
Expand map_functions.R into multiple files

calebbraun committed:
Clean code, fix warnings, bugfix

calebbraun committed:
Fix extent issue with plot_GCAM_grid

calebbraun committed:
Touch-up examples, adjust how zoom works

calebbraun committed:
Add options for graticule placement

calebbraun committed:
Speed up check for invalid polygons

calebbraun committed:
Update tests and documentation

calebbraun committed:
Fix gridded data bounds

rplzzz reviewed (changes requested):
This all looks good.  My one small complaint is that the default plot color for no-data regions is very dark.  This might be ok for the situation where you just want to draw the region boundaries, but more commonly we use missing values to filter data to just some selected regions of interest.  In those cases, we want the suppressed regions to be de-emphasized, which means they need to be a fairly neutral color.  I guess one thing we could do would be to make the color for the continental background settable the same way the color for the ocean background is, but even then, the more neutral color should be the default.

calebbraun commented:
Just to clarify, the dark grey only is the default color when there is no data given by the user, which probably would only be used to show the region boundaries. If there is data (meaning the `col` parameter is supplied), then the `ggplot2` default `NA` color is used and can be set by the user if they add a custom scale:
```
gcamchina[!is.na(gcamchina$id) & gcamchina$id <= 32, 'value'] <- NA
plot_GCAM(map.chn, col='value', gcam_df=gcamchina, proj=ch_aea, extent=EXTENT_CHINA)
```
![china_and_na](https://user-images.githubusercontent.com/12344924/39154597-d230346c-471c-11e8-8b40-27c7aafee734.png)

The reason I initially made the no data maps dark was to increase the contrast between the regions and the plot's panel background color. Too dark though? Thoughts?

rplzzz commented:
Got it.  In that case, the dark color is fine.  I think the region boundaries still don't show up very well against the dark background.  Maybe we should up the value a bit on the light line color that we use on dark backgrounds.

calebbraun committed:
Make no-data map borders lighter

calebbraun committed:
Increase required sf version

They recently fixed an issue where warnings from checking the validity of polygons
could not be suppressed.
